src/psyclone/psyir/symbols/containersymbol.py

```diff
@@ -53,21 +53,48 @@ class ContainerSymbol(Symbol):
 
     :param str name: name of the symbol.
     :param visibility: the visibility of the symbol.
-    :type scope: :py:class:`psyclone.psyir.symbols.Symbol.Visibility`
-
+    :type visibility: :py:class:`psyclone.psyir.symbols.Symbol.Visibility`
+    :param interface: optional object describing the interface to this \
+        symbol (i.e. whether it is passed as a routine argument or \
+        accessed in some other way). Defaults to \
+        :py:class:`psyclone.psyir.symbols.FortranModuleInterface`
+    :type interface: :py:class:`psyclone.psyir.symbols.symbol.SymbolInterface`
     '''
-    def __init__(self, name, visibility=Symbol.DEFAULT_VISIBILITY):
-        # At the moment we just have one ContainerSymbol interface, so we
-        # always assign this interface to all ContainerSymbols. We will
-        # pass the interface as a parameter when we have more than one.
-        super(ContainerSymbol, self).__init__(
-            name, visibility=visibility, interface=FortranModuleInterface())
+    def __init__(self, name, visibility=Symbol.DEFAULT_VISIBILITY,
+                 interface=None):
+        super(ContainerSymbol, self).__init__(name, visibility=visibility)
+
+        if interface is None:
+            # By default it is a FortranModuleInterface
+            self.interface = FortranModuleInterface()
+        elif isinstance(interface, FortranModuleInterface):
```

Comment:
I'm a bit confused here. I thought the interface specified the way in which a 'non-local' symbol is brought into scope/accessed. Is this saying that the ContainerSymbol represents a FortranModule (which seems like a slightly different concept)?

Reply:
I agree this may be confusing (there is an issue to improve the interface names that can help). But in this case I didn't change the behavior of this interface (note that in the previous code FortranModuleInterface was already assigned). I just converted it from a hardcoded value in the call to the super initialization to an argument parameter that just accepts this value.

The reason to transform this to an argument is for the generic `copy()` operation to work as expected with the interface (since `interface` is a property of the generic class I thought the generic `copy` should be in charge of coping it when appropriate)

Reply:
Ah I see the issue now, it is more that a naming issue, I created its own issue and added a TODO for this.

Reply:
Great, thanks.